### ISSUE TITLE
Needs instructions to run

### ISSUE THREAD

**Issue by chrisli30** [TOMOCOIN/performance-test-bot#1]
Hi guys, how do I set up this bot to do perf load test?
A short instruction would be great. Thanks.

**Comment by thanhson1085** [comment]
You should install Golang Compiler in your machine, custom private keys and compile the source code. After that, you can run the test by using binary file.

Now we focus on tomochain, governance dapp, blockexplorer. We will update the instruction in the next time. Thank you

**Comment by chrisli30** [comment]
Thanks Nguyen, I've compiled and run the binary bot. Traces are as below but I don't see any transaction generated from https://stats.tomocoin.io/. Is my output right? My goal is to generate 1000+ tps and see it on https://stats.tomocoin.io/

**Chriss-MacBook-Pro-4:performance-test-bot kezjo$ make**
go build -o bot *.go

**Chriss-MacBook-Pro-4:performance-test-bot kezjo$ ls**
Dockerfile	README.md	collector.go	main.go		sender.go	work.go
Makefile	bot		dispatcher.go	p2p.go		test.sh		worker.go

**Chriss-MacBook-Pro-4:performance-test-bot kezjo$ ./bot** 
Created new account address 0x3B46a158f0959E17379FB932A4eA5e3c5a895621
NodeId 383cce92-76dd-485a-bee9-332b07dc7534
{"address":"3b46a158f0959e17379fb932a4ea5e3c5a895621","crypto":{"cipher":"aes-128-ctr","ciphertext":"d2e6887476769168ea161ab8913ebaa63e3dc4a518daf65fd38233d05a71be80","cipherparams":{"iv":"4c3c6363296ea2adeb931c8623933553"},"kdf":"scrypt","kdfparams":{"dklen":32,"n":262144,"p":1,"r":8,"salt":"bb987b0c4c08edab722c58a323897413e30c1f11a6fb6bcaf1b60d624786dd00"},"mac":"cdfd7591599f4c161f0e9fc9a14a8f9dda3bc0e2a201664d8bd6886f0e5d27e4"},"id":"99f016a5-4a4b-42e6-8d7e-59e67b4d5390","version":3}

./bot -n 1 -url https://core.tomocoin.io -req 1 -port 30304 -bootnodes enode://a19fd515c0429e7b844e35ba5c7b3fc88361ff8793811d005f3aeed7ff4a7b28e367142c4132f7c7e74c2e140cf6fa3ef134997b01ee78b1e613f9435f13dbda@127.0.0.1:30303 -key-file key1.json

**Comment by thanhson1085** [comment]
The bot created account `0x3B46a158f0959E17379FB932A4eA5e3c5a895621` with empty balance (0 $TOMO). So you are not able to make a tx from that account. You should get some free $TOMO from https://faucet.tomocoin.io before re-running the bot.

**Comment by chrisli30** [comment]
I've acquired 3 ETH from testnet faucet in address 0x3B46a158f0959E17379FB932A4eA5e3c5a895621, imported it in main.go and re-make ./bot, but it's still not generating transaction.

I changed line
```
newAcc, err := ks.NewAccount("")
```
to
```
byteArray := []byte(`{"address":"3b46a158f0959e17379fb932a4ea5e3c5a895621","crypto":{"cipher":"aes-128-ctr","ciphertext":"6e8c60b0af2378c6413b4694f43a68ab5d6feed1c482ff60eac128ede7f8afb8","cipherparams":{"iv":"9450d3c64ad792ca2484b448abd0fe6c"},"kdf":"scrypt","kdfparams":{"dklen":32,"n":262144,"p":1,"r":8,"salt":"e56b72c771d123e610fed85203f2f93b350006b5724b7f9b7b1dbd7c88af2472"},"mac":"a5404c7ed6d21dd7825f0b6d6303e5aacf331103dbb5cfe64fc45e1f6be90f0e"},"id":"99f016a5-4a4b-42e6-8d7e-59e67b4d5390","version":3}`)
newAcc, err := ks.Import(byteArray,"qweASD","qweASD")
```
Any idea why? Could you use the key json to give it a test?

Thanks.

**Comment by thanhson1085** [comment]
If you already had the account, you should use `key.json` file. After that, run bot:
```
./bot -n 1 -url https://core.tomocoin.io -key-file key.json
```

**Comment by chrisli30** [comment]
Okay, I've figured it out. Since key file name is hardcoded it has to be "key.json", and other two params needed are attack=1 and req=1000(number of attacks). Saving key json into key.json at root then I used below command to generate transactions.

./bot -attack 1 -req 10000 -n 10

Result:
<img width="861" alt="perf-test" src="https://user-images.githubusercontent.com/2616844/38709960-aa727dfc-3e72-11e8-87dc-c4a9175b685f.png">

Thanks @thanhson1085 !
